dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -790,15 +837,37 @@ HRESULT WINAPI CDesktopFolder::GetDetailsOf(
         psd->cxChar = DesktopSFHeader[iColumn].cxChar;
         return SHSetStrRet(&psd->str, DesktopSFHeader[iColumn].colnameid);
     }
+    else
+    {   /* iColumn is the listview column number in explorer */
+        /* Swap the ReactOS size and type columns */
+        switch (iColumn)
+        {
+            case 0:                /* name */
+            case 3:                /* modified date */
+            case 4:                /* attributes */
+                break;
+            case 1:                /* size */
+                if (_ILIsSpecialFolder(pidl))
+                    iColumnModified = 5;
+                else
+                    iColumnModified = 2;
+                break;
+            case 2:                /* type */
+                iColumnModified = 1;
+        }
+    }
```

Comment:
This comment is misleading. Without going too deep into details, a better comment would be along the lines of "iColumn is the column number of **our** columns the view is requesting information about". But having any comment here at all is superfluous. Explorer is only supposed to know about your (the IShellFolders) columns from what you tell it when Explorer calls GetDetailsOf with a NULL PIDL (and when it does, the folder returns information from its DesktopSFHeader).

Explorer is not supposed to know that 3 is "modified date" and 5 is "comments". Trying to please Explorer by swapping numbers around until it looks correct is not the way to go even though it probably looks like it did the trick.

CDesktopFolder is a merged view of the the two desktop directories and calls that forward to psf (CFSFolder) should keep the column number unchanged because there is a bit of a leaky abstraction here and CDesktopFolder and CFSFolder must keep their columns in sync (unless you want to use MapColumnToSCID instead).

This PR changes DesktopSFHeader to have the first 4 columns in the canonical Windows 95 order (great) and if there is a problem then it is because GenericSFHeader in CFSFolder is still incorrect. The whole `switch `that plays with `iColumnModified` here needs to go away.

Reply:
Code now revised with your help. Thanks.